accounts/abi/bind/backends/simulated.go

```diff
@@ -132,10 +132,23 @@ func (b *SimulatedBackend) CodeAt(ctx context.Context, contract common.Address,
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
-	if blockNumber != nil && blockNumber.Cmp(b.blockchain.CurrentBlock().Number()) != 0 {
-		return nil, errBlockNumberUnsupported
+	var statedb *state.StateDB
```

Comment:
Please put this snippet into a method (`stateForBlockNumber` maybe) so it doesn't need to be repeated four times.